lnwire/extra_bytes.go

```diff
@@ -63,6 +63,23 @@ func (e *ExtraOpaqueData) Decode(r io.Reader) error {
 	return nil
 }
 
+// ValidateTLV checks that the raw bytes that make up the ExtraOpaqueData
+// instance are a valid TLV stream.
+func (e *ExtraOpaqueData) ValidateTLV() error {
+	tlvStream, err := tlv.NewStream()
+	if err != nil {
+		return err
+	}
+
+	// Ensure that the TLV stream is valid by attempting to decode it.
+	_, err = tlvStream.DecodeWithParsedTypesP2P(bytes.NewReader(*e))
```

Comment:
Not sure if we should add an early return in case `e` is actually `nil`? Just to be super defensive...

Reply:
cool yes good idea! will add

Reply:
done!